### ISSUE TITLE
zfsonlinux meaning of signed-off-by?

### ISSUE THREAD

**Issue by inkdot7** [zfsonlinux/zfs#6182]
[`CONTRIBUTING.md`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md) states that each commit should have a `Signed-off-by` line, but does not explain what it would mean?  Searching the net turns up a [stackoverflow question](https://stackoverflow.com/questions/1962094/what-is-the-sign-off-feature-in-git-for), which suggests the meaning being specified in e.g. `CONTRIBUTING.md`.

The requirement for the `Signed-off-by` line in `CONTRIBUTING.md` was added in #5943.

**Comment by bunder2015** [comment]
[Excerpt](http://www.macmillandictionary.com/dictionary/british/sign-off) from the Macmillan dictionary:

> business to officially approve of something or to officially agree that something has been completed in a satisfactory way by signing your name

**Comment by inkdot7** [comment]
@bunder2015  That definition is ambiguous to me:

> to officially approve of something

that may be the meaning that one has permission / the right to release it under the appropriate license

> or to officially agree that something has been completed in a satisfactory way

more sounds like one approves of what the commit actually does

**Comment by kpande** [comment]
"By making a contribution to this project, I certify that:

*    The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or
*    The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless  I am permitted to submit under a different license), as indicated in the file; or
    The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.
*    I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved."

We have the same requirements for using the signed-off-by process as the Linux kernel.

In short, you need to include a signed-off-by tag in every patch:

*    Signed-off-by: this is a developer's certification that he or she has the right to submit the patch for inclusion into the project. It is an agreement to the Developer's Certificate of Origin (above). Code without a proper signoff cannot be merged into the mainline. 

If you are unfamiliar with this process, you should read the [official policy at kernel.org](http://www.kernel.org/doc/Documentation/SubmittingPatches).

**Comment by inkdot7** [comment]
@kpande The link provided is semi-broken / a redirection.

Would it not be suitable to provide a pointer to that document in `CONTRIBUTING.md` ?

**Comment by kpande** [comment]
@inkdot7 signed-off-by is common git nomenclature and this is the meaning. but I don't see why we couldn't add it to `CONTRIBUTING`.

**Comment by kpande** [comment]
use [this link](https://www.kernel.org/doc/html/latest/process/submitting-patches.html)
